libs/server/StoreWrapper.cs

```
@@ -415,6 +416,49 @@ async Task CompactionTask(int compactionFrequencySecs, CancellationToken token =
             }
         }
 
+        async Task HashCollectTask(int hashCollectFrequencySecs, CancellationToken token = default)
+        {
+            Debug.Assert(hashCollectFrequencySecs > 0);
+            try
+            {
+                var scratchBufferManager = new ScratchBufferManager();
+                using var storageSession = new StorageSession(this, scratchBufferManager, null, null, logger);
+
+                while (true)
+                {
+                    if (token.IsCancellationRequested) return;
+
+                    if (objectStore is null)
+                    {
+                        logger?.LogWarning("HashCollectFrequencySecs option is configured but Object store is disabled. Stopping the background hash collect task.");
+                        return;
+                    }
+
+                    ExecuteHashCollect(scratchBufferManager, storageSession);
+
+                    await Task.Delay(TimeSpan.FromSeconds(hashCollectFrequencySecs), token);
+                }
+            }
+            catch (TaskCanceledException ex) when (token.IsCancellationRequested)
```

Token cancellation does not mean we've reached an error state, it just means that the store wrapper was disposed

---

What do you suggest to log in that case? Just suppress the exception?

---

For now, I have suppressed

---

Yup suppressing is fine